woocommerce/woocommerce-android
# [Product AI v2 M1] New preview screen

Closes: #11927 


### Description
This PR adds a new screen showing the product preview during the AI creation.

The PR is quite big, but the review should be simple I hope, the cause of the large changes is the file `AiProductPreviewScreen`, and its content was mostly copied from the old implementation in `ProductPreviewSubScreen`, so it can be skipped.

### Testing information
1. Open the app and make sure you are signed using a Jetpack website.
2. Open product list.
3. Tap on Add a new product.
4. Select "Create a product with AI"
5. Enter any text on the field.
6. Click on "Generate Product Details"
7. Confirm the preview screen is shown with some sample data.

### Images/gif
<img src="https://github.com/woocommerce/woocommerce-android/assets/1657201/cc27ed6c-352c-4aac-929b-3daed69641aa" width=400 />
